Examples/Tests/Macroscopic_Maxwell/inputs_3d_surfaceIntegralDiagnostic_tests

```diff
@@ -0,0 +1,141 @@
+# This is test case for verifying reduced diagnostics to calculate surface/volume integrals for noncubic cells in a noncubic domain. (use USE_LLG=FALSE, ie., no M field)
+# Efield is simulated in a vacuum and initialized on each cells to be equal to 1.
+
+################################
+####### GENERAL PARAMETERS ######
+#################################
+max_step = 500
+amr.n_cell = n_cellx n_celly n_cellz
+
+amr.max_grid_size = 4
+amr.blocking_factor = 8
+geometry.dims = 3
+geometry.prob_lo     =  -Lx/2. -Ly/2. -Lz/2.
+geometry.prob_hi     =   Lx/2.  Ly/2.  Lz/2.
+boundary.field_lo = periodic periodic periodic
+boundary.field_hi = periodic periodic periodic
+amr.max_level = 0
+
+# domain size
+my_constants.n_cellx = 8
+my_constants.n_celly = 64
+my_constants.n_cellz = 24
+
+# grid spacing
+my_constants.Lx = 8.e-6
+my_constants.Ly = 16.e-6
+my_constants.Lz = 12.e-6
+my_constants.tiny = 1.e-12
+my_constants.dx = Lx/n_cellx
+my_constants.dy = Ly/n_celly
+my_constants.dz = Lz/n_cellz
+#################################
+############ NUMERICS ###########
+#################################
+warpx.verbose = 1
+warpx.use_filter = 0
+warpx.cfl = 0.9
+my_constants.sigma = 0.
+my_constants.eps_r = 1.0
+my_constants.mu_r = 1.0
+
+algo.em_solver_medium = macroscopic # vacuum/macroscopic
+algo.macroscopic_sigma_method = laxwendroff # laxwendroff or backwardeuler
+macroscopic.sigma_function(x,y,z) = "sigma"
+macroscopic.epsilon_function(x,y,z) = "8.8541878128e-12*eps_r"
+macroscopic.mu_function(x,y,z) = "1.25663706212e-06*mu_r"
+
+#################################
+############ FIELDS #############
+#################################
+
+my_constants.pi = 3.14159265359
+my_constants.c = 299792458.
+
+warpx.E_ext_grid_init_style = parse_E_ext_grid_function
+warpx.Ez_external_grid_function(x,y,z) = "1"
+warpx.Ex_external_grid_function(x,y,z) = "1"
+warpx.Ey_external_grid_function(x,y,z) = "1"
+
+warpx.B_ext_grid_init_style = parse_B_ext_grid_function
+warpx.Bx_external_grid_function(x,y,z)= "0."
+warpx.By_external_grid_function(x,y,z)= "0."
+warpx.Bz_external_grid_function(x,y,z) = "0."
+
+# Diagnostics
+diagnostics.diags_names = plt chk
+plt.intervals = 10
+plt.fields_to_plot = Ex Ey Ez Bx By Bz
+plt.diag_type = Full
+
+chk.format = checkpoint
+chk.intervals = 100
+chk.diag_type = Full
+
+
+
+######################### Surface Integral on the Y-Plane #############################
+my_constants.eps=1.e-9
+
+warpx.reduced_diags_names = Int_yPlane
+Int_yPlane.type = RawEFieldReduction
+Int_yPlane.reduction_type = integral
+Int_yPlane.integration_type = surface
+Int_yPlane.intervals = 1
+Int_yPlane.surface_normal = Y
```

Comment:
```suggestion
Int_yPlane.surface_normal = Y
```
Does this always have to be upper-case? Also, same as a comment before, what about negative y surface normal?
or is it left to the user to make this negative in the postprocessing when adding the reduced values around an embedded box. If so, could you please elaborate that in the documentation? Thanks!

Reply:
I accounted to lower and upper cases. In the documentation specified that user have to define x, y, z. 
In the code, x or X are accepted.
The discussion about sign is added in the doc.